### PR TITLE
[IA-2072] Add KubernetesService.getServiceExternalIp method

### PR DESCRIPTION
**Pull request by rtitle**
Used by Leo (https://github.com/DataBiosphere/leonardo/pull/1565) to poll whether the nginx ingress controller is up.

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala [hunk 1]
```diff
@@ -1,4 +1,5 @@
 package org.broadinstitute.dsde.workbench.google2
```
Comment (rtitle) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala:1]: The models in this file scare me a little

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/KubernetesInterpreter.scala [hunk 2]
```diff
@@ -144,6 +144,39 @@ class KubernetesInterpreter[F[_]: Async: StructuredLogger: Effect: Timer: Contex
...
+                                    namespace: KubernetesNamespace,
+                                    serviceName: ServiceName)(
+    implicit ev: ApplicativeAsk[F, TraceId]
+  ): F[Option[ServiceExternalIp]] =
+    for {
+      traceId <- ev.ask
+      client <- blockingF(getClient(clusterId, new CoreV1Api(_)))
+      call = blockingF(
+        Async[F].delay(
+          client.listNamespacedService(namespace.name.value, null, "true", null, null, null, null, null, null, null)
```
Comment (Qi77Qi) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/KubernetesInterpreter.scala:156]: should we recover from 404 here as well and return `None`?

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala [hunk 3]
```diff
@@ -428,6 +428,8 @@ object KubernetesModels {
   final case class PortName(value: String) extends AnyVal
   final case class Protocol(value: String) extends AnyVal
 
+  final case class ServiceExternalIp(value: String) extends AnyVal
```
Comment (Qi77Qi) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala]: what does this IP look like? if it's just regular IP, can we just use an `IP` data type?

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/KubernetesInterpreter.scala (hunk 2 above)
Comment (rtitle) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/KubernetesInterpreter.scala:156]: Yeah I think that makes sense

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala (hunk 3 above)
Comment (rtitle) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala]: It's just a regular IP -- good point maybe can just use `IP` data type. I think Leo has an `IP` class, maybe can move that to wb-libs.

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala (hunk 1 above)
Comment (kyuksel) [reply google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/kubernetesModels.scala:1]: me too 😨

**Review by kyuksel — APPROVED** [review]
LGTM 👍

**Pull request merged**